tests/pass-dep/shims/libc-misc.rs

```diff
@@ -213,6 +213,62 @@ fn test_posix_gettimeofday() {
     assert_eq!(is_error, -1);
 }
 
+fn test_localtime_r() {
+    use std::env;
+    use std::ffi::{CStr, CString};
+
+    // Set timezone to GMT
+    //let key = "TZ";
+    //env::set_var(key, "GMT");
+    //assert_eq!(env::var(key), Ok("GMT".to_string()));
+
+    const custom_time: i64 = 1712475836;
+    let custom_time_ptr: *const i64 = &custom_time;
+
+    //TODO: This causes a leak
+    let zone = CString::new("").expect("Cstring new failed").into_raw();
```

Comment:
Miri complains about a memory leak here, what is the proper way to assign a raw pointer for ``*const c_char`` so it will be freed afterwards?